src/library/features/crates/cratetreemodel.h

```diff
@@ -0,0 +1,34 @@
+#ifndef CRATE_TREE_MODEL_H
+#define CRATE_TREE_MODEL_H
+
+#include <QList>
+#include <QMap>
+
+#include "library/treeitemmodel.h"
+#include "library/features/crates/cratemanager.h"
+#include "library/crate/crate.h"
+
+// This class represents the crate tree for the nested crates feature
+
+class CrateTreeModel : public TreeItemModel {
+  public:
+    CrateTreeModel(LibraryFeature* pFeature,
+                   CrateManager* pCrates);
+    //virtual ~CrateTreeModel();
```

Comment:
Please delete unused code instead of commenting it out. But "~CrateTreeModel() override = default;" does not hurt and ensures that we actually have a virtual destructor.